src/pykeen/nn/representation.py

```diff
@@ -985,3 +988,177 @@ def _plain_forward(
+    WIKIDATA_ENDPOINT = "https://query.wikidata.org/bigdata/namespace/wdq/sparql"
+    # image:
+    # https://www.wikidata.org/w/api.php?action=wbgetclaims&property=P18&entity=Qxxx
+
+    def __init__(self) -> None:
+        """Initialize the cache."""
+        self.module = PYKEEN_MODULE.submodule("wikidata")
+
+    @staticmethod
+    def iter_invalid_ids(ids: Sequence[str]) -> Iterable[str]:
+        """Iterate over invalid IDs."""
+        pattern = re.compile(r"Q(\d+)")
+        for one_id in ids:
+            if not pattern.match(one_id):
+                yield one_id
+
+    @staticmethod
+    def _safe_get(d: dict, *keys: str, default=None) -> Optional[Any]:
+        """
+        Get from a nested dictionary with default.
+
+        :param d:
+            the (nested) dictionary
+        :param keys:
+            the sequence of keys
+        :param default:
+            the default value if a key is not present at any level, defaults to None
+        :return: the value, or the default
+        """
+        # TODO: move to utils
+        for key in keys[:-1]:
+            d = d.get(key, {})
+        return d.get(keys[-1], default)
+
+    @staticmethod
+    def query(
+        wikidata_ids: Sequence[str], language: str = "en", batch_size: int = 256
+    ) -> Mapping[str, Mapping[str, str]]:
+        """
+        Query the SPARQL endpoints about information for the given IDs.
+
+        :param wikidata_ids:
+            the Wikidata IDs
+        :param language:
+            the label language
+        :param batch_size:
+            the batch size; if more ids are provided, break the big request into multiple smaller ones
+
+        :return:
+            a mapping from Wikidata Ids to dictionaries with the label and description of the entities
+        """
+        # cf. https://github.com/biopragmatics/bioregistry/blob/55584709e287d1d01d51375e0bd836f3c4d25b2e/src/bioregistry/utils.py#L53-L63  # noqa: E501
+        if not wikidata_ids:
+            return {}
+
+        if len(wikidata_ids) > batch_size:
+            # break into smaller requests
+            return dict(
+                itertools.chain.from_iterable(
+                    WikidataCache.query(wikidata_ids=id_batch, language=language, batch_size=batch_size).items()
+                    for id_batch in more_itertools.chunked(wikidata_ids, batch_size)
+                )
+            )
+
+        # is this already part of the requirements?
+        import requests
+
+        qualified = " ".join(f"wd:{i}" for i in wikidata_ids)
+        sparql = f"""
+            SELECT ?item ?itemLabel ?itemDescription WHERE {{{{
+                VALUES ?item {{ {qualified} }}
+                SERVICE wikibase:label {{ bd:serviceParam wikibase:language "{language}". }}
+            }}}}
+        """
+
+        logger.debug("running query: %s", sparql)
+        res = requests.get(WikidataCache.WIKIDATA_ENDPOINT, params={"query": sparql, "format": "json"})
```

Comment:
we'd rather add a User-Agent, too - otherwise the Wikidata endpoint will ban the script after few automatic queries - been there so many times 😅

Reply:
Mood

Reply:
https://github.com/pykeen/pykeen/pull/966/commits/abc76f2b263ac13cae7de4c95bf666c29bc887a3

Reply:
Done in 05c4ea69